packages/grant-explorer/src/features/common/Navbar.tsx

```diff
@@ -17,27 +15,26 @@ export interface NavbarProps {
 }
 
 export default function Navbar(props: NavbarProps) {
-  const [cart] = useCart();
-  const showWalletInteraction = props.showWalletInteraction ?? true;
-  const currentOrigin = window.location.origin;
-
-  const { roundId } = useParams();
-  const { address: walletAddress } = useAccount();
+  /* This part keeps the store in sync between tabs */
+  const store = useCartStorage();
+  const updateStore = () => {
+    useCartStorage.persist.rehydrate();
+  };
 
   useEffect(() => {
-    if (!roundId) {
-      return;
-    }
-    const storageEventHandler = (event: StorageEvent) =>
-      reloadPageOnLocalStorageEvent(roundId, event);
-
-    window.addEventListener("storage", storageEventHandler);
-
-    // Clean up the event listener when the component unmounts
+    document.addEventListener("visibilitychange", updateStore);
```

Comment:
nope, because the useEffect runs only once - on first render, and then the cleanup function returned from useffect runs on unmount. which means that the ref to the function will remain the same for the one render cycle.

Reply:
But a good question, had to think and consult with gpt4 on this one.